src/services/recim/match.ts

```diff
@@ -0,0 +1,75 @@
+import http from '../http';
+import { Participant } from './participant';
+import { Team } from './team';
+
+export type Match = {
+  ID: number;
+  Time: string;
+  Surface: string;
+  Status: string;
+  SeriesID: number;
+  Attendance: Participant[];
+  Team: Team[];
+};
+
+type CreatedMatch = {
+  ID: number;
+  Time: string;
+  SurfaceID: number;
+  StatusID: number;
+  SeriesID: number;
+};
+
+type UploadMatch = {
+  StartTime: string;
+  SeriesID: number;
+  SurfaceID: number;
+  TeamIDs: Array<number>;
+};
+
+type CreatedMatchStats = {
+  ID: number;
+  TeamID: number;
+  MatchID: number;
+  StatusID: number;
+  Score: number;
+  Sportsmanship: number;
+};
+
+type PatchMatchStats = {
+  TeamID: number;
+  Status: string;
+  Score: number;
+  Sportsmanship: number;
+};
+
+type PatchMatch = {
+  Time: string;
+  SurfaceID: number;
+  StatusID: number;
+};
+
+type CreatedAttendance = {
+  ID: number;
+  MatchID: number;
+  ParticipantID: number;
+};
+
+//Match Routes
+const createMatch = (newMatch: UploadMatch): Promise<CreatedMatch> =>
+  http.post('recim/matches', newMatch);
+
+const createMatchAttendance = (username: string, matchID: number): Promise<CreatedAttendance> =>
+  http.post(`recim/matches/attendance?matchID=${matchID}&username=${username}`);
```

Comment:
Data in a created object should go in a post body rather than the URL GET parameters. Also, the match identifier should be a part of the URI path, not a GET/POST parameter.
```suggestion
const createMatchAttendance = (username: string, matchID: number): Promise<CreatedAttendance> =>
  http.post(`recim/matches/${matchID}/attendance`, username);
```

Reply:
since username is a string, i think asp.net automatically puts it in the query. I could switch it to FromBody but regardless it might be a bit odd just to have a single string there.

Reply:
ASP.NET checks all potential places where data could reside. If it does not find it in the query string, then it will check the post body.

If you are posting data, it should be in the body, even if it is just a string. "Query" parameters are called query parameters because their primary purpose is to filter a dataset, not to pass data to the endpoint.